storage/interface.go

```diff
@@ -158,13 +158,13 @@ func (f QueryableFunc) Querier(ctx context.Context, mint, maxt int64) (Querier,
 // Operations on the Appender interface are not goroutine-safe.
 type Appender interface {
 	// Append adds a sample pair for the given series.
-	// An optional reference number can be provided to accelerate calls.
-	// A reference number is returned which can be used to add further
-	// samples in the same or later transactions.
-	// Returned reference numbers are ephemeral and may be rejected in calls
```

Comment:
> Returned reference numbers are ephemeral and may be rejected in calls
 to Append() at any point
 
 What does it mean? Does it mean if I put ref across calls my Append can reject it (I did not see any implementation doing this). Does it mean we always need to provided labels too? (I think we need, so worth mentioning). Or does it mean ref can suddenly point to wrong series across single commit?

Reply:
> What does it mean? Does it mean if I put ref across calls my Append can reject it (I did not see any implementation doing this).

Between new head blocks the references can change I believe, although I could be wrong.

> Does it mean we always need to provided labels too?

Yes.

> Or does it mean ref can suddenly point to wrong series across single commit?

I actually think there is a bug in this (and the old) behaviour where this could be true...

Reply:
Well it should not happen in practice. I expect that in case you have a wrong ref, tsdb would return you err.NotFound, which we handled gracefully I think.